core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableJoin.java

```diff
@@ -227,6 +223,40 @@ public Result implement(EnumerableRelImplementor implementor, Prefer pref) {
                             joinType.generatesNullsOnRight())))).toBlock());
   }
 
+  Expression predicate(EnumerableRelImplementor implementor,
```

Comment:
Small detail to keep consistency, I think it would be better to use an action (a verb) in the method name, e.g. something like `generatePredicate`

Reply:
Agree with you.

Reply:
done.